### PR TITLE
Update the postCommit hook godoc to reflect API

### PR DESCRIPTION
**Pull request by smarterclayton**
Formatting changed to read better in generated docs.

@bparees for 1.2 for docs

### CODE REVIEW

**Comment by smarterclayton** [comment]
[merge]

File: pkg/build/api/v1/types.go [hunk 1]
```diff
@@ -417,53 +417,53 @@ type SourceBuildStrategy struct {
 //
 // 1. Shell script:
 //
-// 	BuildPostCommitSpec{
-// 		Script: "rake test --verbose",
-// 	}
+// 	      "postCommit": {
+//          "script": "rake test --verbose",
```
Comment (bparees) [inline pkg/build/api/v1/types.go]: indent?

File: pkg/build/api/v1/types.go [hunk 2]
```diff
@@ -417,53 +417,53 @@ type SourceBuildStrategy struct {
...
-// The above is a convenient form which is equivalent to:
+//     The above is a convenient form which is equivalent to:
 //
-// 	BuildPostCommitSpec{
-// 		Command: []string{"/bin/sh", "-ic"},
-// 		Args: []string{"rake test --verbose"},
-// 	}
+// 	      "postCommit": {
+// 		      "command": ["/bin/sh", "-ic"],
+// 	        "args":    ["rake test --verbose"]
```
Comment (bparees) [inline pkg/build/api/v1/types.go]: align indents?

File: pkg/build/api/v1/types.go [hunk 3]
```diff
@@ -417,53 +417,53 @@ type SourceBuildStrategy struct {
...
+//     Command overrides the image entrypoint in the exec form, as documented in
+//     Docker: https://docs.docker.com/engine/reference/builder/#entrypoint.
 //
 // 3. Pass arguments to the default entrypoint:
 //
-// 	BuildPostCommitSpec{
-// 		Args: []string{"rake", "test", "--verbose"},
-// 	}
+//        "postCommit": {
+// 		      "args": ["rake", "test", "--verbose"]
```
Comment (bparees) [inline pkg/build/api/v1/types.go]: match w/ other indenting

**Comment by bparees** [comment]
indenting seems messed up, otherwise lgtm

File: pkg/build/api/v1/types.go (hunk 3 above)
Comment (smarterclayton) [inline pkg/build/api/v1/types.go]: Crap, stupid editor.

**Comment by smarterclayton** [comment]
The checked in code has no tabs, but GitHub seems to think there are.

**Comment by smarterclayton** [comment]
[merge]

File: pkg/build/api/v1/types.go (hunk 3 above)
Comment (smarterclayton) [inline pkg/build/api/v1/types.go]: GitHub is wrong, the actual code has no tabs.

Comment (smarterclayton) [inline pkg/build/api/v1/types.go]: Oo... Git was broken.  Impressive.

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**